app/libs/soundcloud.js

```diff
@@ -24,83 +25,189 @@ class Soundcloud {
                     type: 'playlist'
                 });
                 application.allPlaylists.create(playlist);
-                for (let i = 0; i < res.tracks.length; i++) {
-                    let track = res.tracks[i];
-                    this.checkIfAlreadyExist(track, playlist);
-                }
+                this.launchImport(res.tracks, playlist);
             } else if (res.kind == 'track') {
-                this.checkIfAlreadyExist(res);
+                this.launchImport([ res ]);
             }
         });
     }
 
+    // Create a promise for each track
+    launchImport(tracks, playlist) {
+        let promises = [];
+        for (let i = 0; i < tracks.length; i++) {
+            let track = tracks[i];
+            track.playlist = playlist;
+            let promise = new Promise((resolve, reject) => {
```

Comment:
Using Promise constructor is almost always an anti-pattern.
In this case, you can do something like
```
if(!track.streamable){
     promise = Promise.resolve(.... not streamable)
}else{
       promise = this.checkAlreadyExists(track)
}
``` 

You will need to change checkAlreadyExist to return a Promise.
The global idea, is that mixin Promise & callbacks make it weird, use them everywhere (from cozy-browser-sdk) or dont use them.